apps/anoma_node/lib/node/utility/consensus.ex

```diff
@@ -3,54 +3,75 @@ defmodule Anoma.Node.Utility.Consensus do
   A trivial consensus provider for the mempool.
   """
 
-  alias Anoma.Node.Transaction.Mempool
   alias __MODULE__
+  alias Anoma.Crypto.Id
+  alias Anoma.Node.Transaction.Mempool
+  alias Anoma.Node.Registry
 
   use TypedStruct
 
   typedstruct do
     field(:interval, non_neg_integer(), default: 5000)
+    field(:node_id, Id.t())
   end
 
-  def start_link(interval \\ 5000) do
-    GenServer.start_link(__MODULE__, interval, name: Consensus)
+  def start_link(args) do
+    args = Keyword.validate!(args, [:interval, :node_id])
+    name = Registry.name(args[:node_id], __MODULE__)
+    GenServer.start_link(__MODULE__, args, name: name)
   end
 
   @spec init(any()) :: {:ok, Consensus.t()}
-  def init(interval) do
+  def init(args) do
+    Process.set_label(__MODULE__)
+    args = Keyword.validate!(args, [:interval, :node_id])
+
     EventBroker.subscribe_me([
       block_filter()
     ])
 
+    table =
+      String.to_atom(
+        "#{Anoma.Node.Logging.Events}_#{:erlang.phash2(args[:node_id])}"
+      )
+
     :mnesia.transaction(fn ->
-      for cons <- :mnesia.read(Anoma.Node.Logging.Events, :consensus) do
+      for cons <- :mnesia.read(table, :consensus) do
+        IO.puts("wait for block")
```

Comment:
```suggestion
```

Reply:
Don't fix and re-roll everything just yet though. I have an already prepared branch with the fix we can make into a PR on top of everything. Might be easier

Reply:
Alright, I'll keep it in then.